src/routers/GalleryRouter.ts

```diff
@@ -60,48 +66,48 @@ module.exports = ({ $gallery, $mongo: { $db } }: ContextType) => {
   });
 
   router.get('/gallery/file/:id/:name', cache({ ttl: sixMonths }), (req: any, res) => {
-    const { id, utm_medium, utm_campaign, utm_content } = req.params;
-    return trackEvent(utm_medium, utm_campaign, utm_content).then(() => {
-      return $gallery
-        .fetchOriginal({ itemId: id })
-        .then((fileBuffer: any) => {
-          if (!fileBuffer) return res.status(500).send('File not found');
-
-          return res.type(fileBuffer.ContentType).send(fileBuffer.Body);
-        })
-        .catch((err: any) => {
-          res.status(500).send(err);
-        });
-    });
+    const { id } = req.params;
+    return $gallery
+      .fetchOriginal({ itemId: id })
+      .then((fileBuffer: any) => {
+        if (!fileBuffer) return res.status(500).send('File not found');
+
+        return res.type(fileBuffer.ContentType).send(fileBuffer.Body);
+      })
+      .catch((err: any) => {
+        res.status(500).send(err);
+      });
   });
 
   router.get('/gallery/file/:id', (req, res) => {
     const { id } = req.params;
 
-    return $db
-      .collection('gallery')
-      .findOne<GalleryItem>({ _id: id })
-      .then(item => {
-        res.redirect(`/gallery/file/${id}/${item?.fileInfo?.originalname}?utm_medium=website&utm_campaign=website&utm_content=${item?.fileInfo?.originalname}`);
-      });
+    return $mongo.then(({ $db }) => {
+      return $db
+        .collection('gallery')
+        .findOne<GalleryItem>({ _id: id })
+        .then(item => {
+          res.redirect(`/gallery/file/${id}/${item?.fileInfo?.originalname}`);
+        });
+    });
   });
 
   return router;
 };
 
-const trackEvent = (medium: string, campaign: string, fileName: string) => {
-  const trackingId = process.env.GA_TRACKING_ID;
-  if (!trackingId || !trackingId.length) return Promise.resolve();
-  if (!medium || !campaign || !fileName) return Promise.resolve();
-  const data = {
-    v: '1',
-    tid: trackingId,
-    cm: medium,
-    cc: fileName,
-    cn: campaign,
-  };
-
-  return fetch('http://www.google-analytics.com/debug/collect', {
-    params: data,
-  });
-};
+// const trackEvent = (medium: string, campaign: string, fileName: string) => {
+//   const trackingId = process.env.GA_TRACKING_ID;
+//   if (!trackingId || !trackingId.length) return Promise.resolve();
```

Comment:
Looks like this function was removed above

Reply:
True, mostly keeping it for the call to GA. Need to move it to its own tracking service